main.go

```diff
@@ -624,16 +638,6 @@ func (s *DB) NewRecord(value interface{}) bool {
 	return s.NewScope(value).PrimaryKeyZero()
 }
 
-// RecordNotFound check if returning ErrRecordNotFound error
-func (s *DB) RecordNotFound() bool {
-	for _, err := range s.GetErrors() {
```

Comment:
A sanity check:

It seems this method was going through all the errors while what we do now is 
errors.Is(s.Error ....). Do we actually miss checking some errors or is errors.Is doing the job for us?

Reply:
There is this helper method as well: https://github.com/spacelift-io/gorm/blob/447d578628011308498d9316838f59f93834967c/errors.go#L25

but yeah, all the unit tests are passing nicely, so it should work fine in this lib.